File: backend/src/main/java/by/minilooth/diploma/service/vehicle/ModelService.java
package by.minilooth.diploma.service.vehicle;

import by.minilooth.diploma.exception.ActionIsImpossibleException;
import by.minilooth.diploma.exception.vehicle.MakeNotFoundException;
import by.minilooth.diploma.exception.vehicle.ModelNotFoundException;
import by.minilooth.diploma.models.bean.vehicle.Make;
import by.minilooth.diploma.models.bean.vehicle.Model;
import by.minilooth.diploma.models.spareparts.vehicle.ProcessModel;

import java.util.List;
import java.util.Optional;

public interface ModelService {

    void save(Model model);
    Model save(ProcessModel processModel);
    Model update(ProcessModel processModel, Long id) throws ModelNotFoundException;
    void delete(Model model);
    Model delete(Long id) throws ModelNotFoundException, ActionIsImpossibleException;
    Model getById(Long id) throws ModelNotFoundException;
    List<Model> getAllByMakeSorted(Long id) throws MakeNotFoundException;
    List<Model> getAllByMake(Long id) throws MakeNotFoundException;
    List<Model> getAll();
    Boolean existsByMake(Make make);
    Boolean existsById(Long id);

}

File: frontend/service/spareparts/vehicle/ModelService.ts
import {Model, ProcessModel} from "types/spareparts/vehicle/model";
import {Axios} from "core/axios";
import {IncomingHttpHeaders} from "http";

enum ModelRoutes {
  ALL = "/all"
}

export class ModelService {

  private static readonly URL = '/model';

  static async getAll(makeId: number | string, headers?: IncomingHttpHeaders): Promise<Model[]> {
    const {data: models} = await Axios.get<Model[]>(`${this.URL + ModelRoutes.ALL}/${makeId}`, {headers});
    return models;
  }

  static async getById(id: number | string, headers: IncomingHttpHeaders = {}): Promise<Model> {
    const {data: model} = await Axios.get<Model>(`${this.URL}/${id}`, {headers});
    return model;
  }

  static async add(processModel: ProcessModel): Promise<Model> {
    const {data: model} = await Axios.put<Model>(this.URL, processModel);
    return model;
  }

  static async update(id: number | string, processModel: ProcessModel): Promise<Model> {
    const {data: model} = await Axios.patch<Model>(`${this.URL}/${id}`, processModel);
    return model;
  }

  static async delete(id: number | string): Promise<Model> {
    const {data: model} = await Axios.delete<Model>(`${this.URL}/${id}`);
    return model;
  }

}
